src/odemis/gui/model/__init__.py

```diff
@@ -288,6 +289,9 @@ def __init__(self, microscope):
 
                 # Otherwise, just not interested by this component
 
+            #If the state of an HW component changes create an OS pop up message
+            observe_comp_state(comps_with_role)
```

Comment:
As mentioned before, it's probably best to listen to *every* component. As the ones with roles can be some simple wrapper on top of the real hardware, which will have an error.

You probably can reuse the `model.getComponents()` and instead of using it directly in the loop, save it in a variable.
Also, note that the function will need to hold a reference to the VA (or the component itself), as they will be dereferenced if you don't. (I guess it works for now because the SettingsObserver already holds a reference to each of the components with role).